Mathlib/Combinatorics/SimpleGraph/Connectivity/WalkCounting.lean

```diff
@@ -184,6 +184,22 @@ lemma odd_card_iff_odd_components : Odd (Nat.card V) ↔
   rw [Nat.card_eq_fintype_card, Fintype.card_ofFinset]
   exact (Finset.odd_sum_iff_odd_card_odd (fun x : G.ConnectedComponent ↦ Nat.card x.supp))
 
+lemma ConnectedComponent.odd_card_supp_iff_odd_subcomponents {G'} [DecidableRel G.Adj]
+    [DecidableRel G'.Adj] (h : G ≤ G') (c' : ConnectedComponent G') :
+    Odd (Nat.card c'.supp) ↔ Odd (Nat.card
+    ({(c : {c : ConnectedComponent G | c.supp ⊆ c'.supp}) | Odd (Nat.card c.1.supp) })) := by
```

Comment:
Any reason why you are nesting set comprehensions here?
```suggestion
    {c : ConnectedComponent G | c.supp ⊆ c'.supp ∧ Odd (Nat.card c.1.supp)}) := by
```

Reply:
Mainly because I couldn't figure out how to rewrite between these two forms in using it :see_no_evil:

Reply:
I think you never need your original form. You must have thought you needed it because of your statement of `union_supp_eq_supp`, but my suggestion below should fix this